src/into_elements.rs

```diff
@@ -0,0 +1,65 @@
+use data::{DataMap, Element, FromElements};
+use visit::{EdgeRef, Data, GraphBase,  IntoNodeIdentifiers, IntoEdgeReferences};
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::hash::Hash;
+use std::rc::Rc;
+
+/// Turn a graph with clonable weights that implementings the correct taits into its Elements.
+pub struct IntoElements<'a, G: 'a>(pub &'a G);
+
+
+//Todo: ids do not need to be cloneable. I couldn't get the lifetimes of EdgeWeight to work.
+/// Graphs that have a Cloneable weigths and implement the necessary traits can be deconstructed into elements.
+impl<'a, G> IntoIterator for IntoElements<'a, G>
+    where &'a G: GraphBase + DataMap + IntoNodeIdentifiers + IntoEdgeReferences,
+          <&'a G as Data>::NodeWeight: Clone,
+          <&'a G as Data>::EdgeWeight: Clone,
+          <&'a G as GraphBase>::NodeId: Eq + Hash + Clone,
```

Comment:
Please convert this into using `G` as a GraphRef instead of hardcoding it as `&G`. Same in the struct definition.